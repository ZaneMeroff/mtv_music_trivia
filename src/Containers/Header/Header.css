#header-container {
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  height: 200px;
  background-color: #fff200;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
}

#nav-logo {
  height: 160px;
  width: 190px;
}

#score-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

#question-tally {
  font-size: 180%;
  margin-bottom: -15%;
}

#score-tally {
  font-size: 290%;
}

#correct-answers {
  color: #03a9f4;
  margin-right: 8px;
}

#incorrect-answers {
  color: #ff64de;
  margin-left: 8px;
}

#difficulty-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

#difficulty-heading,
#difficulty-rating {
  font-size: 180%;
}

#difficulty-rating {
  margin-top: -15%;
}
