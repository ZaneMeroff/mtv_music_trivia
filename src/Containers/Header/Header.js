import React, { Component } from 'react';
import { connect } from 'react-redux';
import logo from '../../assets/mtv_logo_pink.png';
import './Header.css';

class Header extends Component {

  render() {
    return (
      <section id='header-container'>
        <img src={ logo } id='nav-logo' alt='mtv trivia logo'/>
        <div id='score-container'>
          <p id='question-tally'>Question {this.props.correct.length + this.props.incorrect.length} of 10</p>
          <p id='score-tally'><span id='correct-answers'> {this.props.correct.length}âœ” </span> <span id='incorrect-answers'> {this.props.incorrect.length}x</span></p>
        </div>
        <div id='difficulty-container'>
          <p id='difficulty-heading'>Difficulty:</p>
          <p id='difficulty-rating'>{this.props.difficulty.toUpperCase()}</p>
        </div>
      </section>

    )

  }

}

export const mapStateToProps = (state) => ({
  difficulty: state.difficulty,
  correct: state.correctQuestions,
  incorrect: state.incorrectQuestions
});

export default connect(mapStateToProps)(Header);
