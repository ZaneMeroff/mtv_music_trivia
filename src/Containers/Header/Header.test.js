import React from 'react';
import { Header, mapStateToProps } from './Header';
import { shallow } from 'enzyme';

  //  * * * * * Get mapStateToProps to pass!

describe('Header', () => {
  let wrapper;
  let question;
  beforeEach(() => {
    question = {question: 'Who da best band?'}
    wrapper = shallow(<Header difficulty={'easy'} correct={question} incorrect={question} />)
  })
  it ('should render a Header component', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return an object with a difficulty, array of correct questions, and an array of incorrect questions', () => {
    const questions = [{question: 'Who da best band?'}]
    const mockState = {
      difficulty: 'easy',
      correctQuestions: questions,
      incorrectQuestions: questions,
    };
    const expected = {
      difficulty: 'easy',
      correctQuestions: questions,
      incorrectQuestions: questions,
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected)
  });
});
