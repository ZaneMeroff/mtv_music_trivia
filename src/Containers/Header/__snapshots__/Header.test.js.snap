// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render a Header component 1`] = `
<section
  id="header-container"
>
  <img
    alt="mtv trivia logo"
    id="nav-logo"
    src="mtv_logo_pink.png"
  />
  <div
    id="score-container"
  >
    <p
      id="question-tally"
    >
      Question 
      NaN
       of 10
    </p>
    <p
      id="score-tally"
    >
      <span
        id="correct-answers"
      >
         
        âœ” 
      </span>
       
      <span
        id="incorrect-answers"
      >
         
        x
      </span>
    </p>
  </div>
  <div
    id="difficulty-container"
  >
    <p
      id="difficulty-heading"
    >
      Difficulty:
    </p>
    <p
      id="difficulty-rating"
    >
      EASY
    </p>
  </div>
</section>
`;
